shared/src/business/entities/JoiValidationEntity.ts

```diff
@@ -192,14 +192,18 @@ export abstract class JoiValidationEntity {
     return this.validate({ applicationContext, logErrors: true });
   }
 
-  static validateRawCollection<T extends JoiValidationEntity>(
-    this: new (someVar: any, someArgs: any) => T,
+  static validateRawCollection<ClassType extends new (...args: any) => any>(
+    this: ClassType,
     collection: any[] = [],
-    args: any,
+    ...args: Tail<ConstructorParameters<ClassType>>
   ) {
     return collection.map(
       rawEntity =>
-        new this(rawEntity, args).validate().toRawObject() as ExcludeMethods<T>,
+        new this(rawEntity, ...args).validate().toRawObject() as ExcludeMethods<
+          InstanceType<ClassType>
+        >,
     );
   }
 }
+
+type Tail<T extends any[]> = T extends [any, ...infer Rest] ? Rest : never;
```

Comment:
Based on the commit comment I assume that this is merely a typing change and not a substantive change. My Typescript-fu is weak: could someone explain the effect that this change has and/or how it improves the developer experience?

Reply:
Ugh.... This function. So there is one minor change to the functionality of this function but everything else is type changes. The functional change is that no validateRawCollection spreads args (...args) so that a constructor can accept more than two arguments.

The typing for this function is, and let's be honest, crazy. It is just crazy. Before we made changes this function was not typed and just accepted any's which meant that if you did not pass in the right constructor arguments you would not be warned. We updated the types and now Typescript will complain if you do not pass in the correct arguments. Hooray! 

The typing to achieve that is pretty ugly but it is honest. Personally I do not think this function should exist because the typing is complex and the functionality is not that helpful. So we only improved the typing because we did not think it was a good idea to undertake a refactoring of validateRawCollection in such a massive PR.